hera_cal/xtalk_filter.py

```diff
@@ -202,52 +216,13 @@ def load_xtalk_filter_and_write_baseline_list(datafile_list, baseline_list, calf
         cals = None
     xf = XTalkFilter(hd, input_cal=cals, round_up_bllens=round_up_bllens)
     xf.read(bls=baseline_list, frequencies=freqs)
+    if factorize_flags:
+        xf.factorize_flags(time_thresh=time_thresh, inplace=True)
     xf.run_xtalk_filter(cache_dir=cache_dir, read_cache=read_cache, write_cache=write_cache, **filter_kwargs)
     xf.write_filtered_data(res_outfilename=res_outfilename, CLEAN_outfilename=CLEAN_outfilename,
                            filled_outfilename=filled_outfilename, partial_write=False,
                            clobber=clobber, add_to_history=add_to_history,
-                           extra_attrs={'Nfreqs': len(freqs), 'freq_array': np.asarray([freqs])})
-
-
-def reconstitute_xtalk_files(templatefile, fragments, outfilename, clobber=False):
-    """Recombine xtalk products into short-time files.
-
-    Construct a new file based on templatefile that combines the files in file_fragments
-    over the times in template file.
-
-    Arguments
-    ---------
-    templatefile : string
-        name of the file to use as a template. Will reconstitute the file_fragments over the times in templatefile.
-    outfilename : string
-        name of the output file
-    file_fragments : list of strings
-        list of file names to use reconstitute.
-    clobber : bool optional.
-        If False, don't overwrite outfilename if it already exists. Default is False.
-
-    Returns
-    -------
-        Nothing
-    """
-    hd_template = io.HERAData(templatefile)
-    hd_fragment = io.HERAData(fragments[0])
-    times = hd_template.times
-    freqs = hd_fragment.freqs
-    polarizations = hd_fragment.pols
-    # read in the template file, but only include polarizations, frequencies
-    # from the fragment files.
-    hd_template.read(times=times, frequencies=freqs, polarizations=polarizations)
-    # for each fragment, read in only the times relevant to the templatefile.
-    # and update the data, flags, nsamples array of the template file
-    # with the fragment data.
-    for fragment in fragments:
-        hd_fragment = io.HERAData(fragment)
-        d, f, n = hd_fragment.read(times=times)
-        hd_template.update(flags=f, data=d, nsamples=n)
-    # now that we've updated everything, we write the output file.
-    hd_template.write_uvh5(outfilename, clobber=clobber)
-
+                           extra_attrs={'Nfreqs': xf.Nfreqs, 'freq_array': np.asarray([xf.freqs])})
```

Comment:
same comment here

Reply:
hopefully it'll withstand the pyuvdata winds.